jax/_src/mesh.py

```diff
@@ -230,6 +230,8 @@ class Mesh(_BaseMesh, contextlib.ContextDecorator):
     axis_names: A sequence of resource axis names to be assigned to the
       dimensions of the ``devices`` argument. Its length should match the
       rank of ``devices``.
+    axis_types: and optional tuple of :class:`jax.sharding.AxisType` entries corresponding to
```

Comment:
Add this to AbstractMesh too below? Also link to https://docs.jax.dev/en/latest/notebooks/explicit-sharding.html in all the `axis_types` doc?